### DISCUSSION TITLE
The Scaler in test.zip predict.py requires train data, but when upload with train data, the file will exceed 200MB

### DISCUSSION THREAD

**Discussion by GuibinChen92057** [PaddlePaddle/PaddleSpatial#248] [BaiduKDDCup2022] [answered]
In the predict.py, the forecast_one func, we need to use the train_data.get_scaler() to transform and inverse_trainsform test_x, but when we put the 'data/wtb_245days.csv' in our submit file, it must exceed 200MB. How can we handle with it?
![image](https://user-images.githubusercontent.com/85916561/178147474-c28c00c1-3842-4ea6-a3f5-d8c5f23a6431.png)

**Comment by shaido987** [comment]
What I'm doing with the scaler is saving it in a pickle file during training and including it in the submitted files. Then the scaler can be loaded and used at prediction time. This has the additional advantage of being faster as well.

**Comment by GuibinChen92057** [comment]
Thank you for your reply, but I still have some question. What if I adopt other "scaler" function? For example, the released version using the scaler defined in the Wind Turbine Data.py file, so we can invoke it faster in our upload predict.py file. But if I define other form of "scaler" (i.e., si-ki learn StandardScaler), then will error happens when uploaded? many thx if you can reply!



---- Replied Message ----
| From | ***@***.***> |
| Date | 07/11/2022 09:23 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [PaddlePaddle/PaddleSpatial] The Scaler in test.zip predict.py requires train data, but when upload with train data, the file will exceed 200MB (Discussion #248) |

What I'm doing with the scaler is saving it in a pickle file during training and including it in the submitted files. Then the scaler can be loaded and used at prediction time. This has the additional advantage of being faster as well.

**Comment by ramber1836** [comment, marked as answer]
@GuibinChen92057 
You can customize the wind_turbine_data.py with your favorite Scaler implementation. 
The backend evaluation does not presume any specific scaler function.
